### ISSUE TITLE
Inconsistency In CategoricalCrossentropy Calculation

### ISSUE THREAD

**Issue by River861** [tensorflow/tensorflow#51720]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  **Yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Windows 10**
- TensorFlow installed from (source or binary): **binary**
- TensorFlow version (use command below):  **2.6.0**
- Python version:  **3.6.13**
- GPU model and memory:  **CPU-only**

**Describe the current behavior**
The `categorical_crossentropy` loss values calculated with different ways is inconsistent in a model built by myself. 

1. In the first way, I calculate the loss value using `model.predict` and `CategoricalCrossentropy`
2. In tne second way, I calculate the loss value using `model.evaluate`
3. In the third way, I show the loss value using `model.fit`

The small model, input data `x`, target data `y_true` and the testing code are here:  [cce_test.zip](https://github.com/tensorflow/tensorflow/files/7068915/cce_test.zip)

The three results are: 
```
14.506287     287.1078     287.1078
```
Besides, I calculate the loss with the same model and data using `CNTK2.7` and Theano`1.0.4`，and their result are both `14.506287`. This makes me more suspicious to the calculation result of tensorflow.

**Describe the expected behavior**
```
14.506287    14.506287     14.506287
```

**Standalone code to reproduce the issue**
```python
import numpy as np
import tensorflow as tf

model = tf.keras.models.load_model("model.h5")
x = np.load("x.npy")
y_true = np.load("y_true.npy")

model.compile(loss='categorical_crossentropy', optimizer='adam')

# calculate loss using model.predict and categorical_crossentropy
cce = tf.keras.losses.CategoricalCrossentropy()
y_pred = model.predict(x=x)
loss1 = cce(y_true, y_pred)
print("loss1 = ", loss1.numpy())

# calculate loss using model.evaluate
loss2 = model.evaluate(x=x, y=y_true)
print("loss2 = ", loss2)

# show loss using model.fit
model.fit(x=x, y=y_true)
```
The result is:
```
loss1 =  14.506287
1/1 [==============================] - 0s 461ms/step - loss: 287.1078
loss2 =  287.10784912109375
1/1 [==============================] - 2s 2s/step - loss: 287.1078
```

**Other info / logs** 
Here is the code do the same thing using CNTK2.7:
```python
import numpy as np
import os
os.environ['KERAS_BACKEND'] = 'cntk'
import keras

model = keras.models.load_model("model.h5")
x = np.load("x.npy")
y_true = np.load("y_true.npy")

model.compile(loss='categorical_crossentropy', optimizer='adam')

# calculate loss using model.predict and categorical_crossentropy
cce = keras.losses.CategoricalCrossentropy()
y_pred = model.predict(x=x)
loss1 = cce([y_true], [y_pred])
print("loss1 = ", loss1.eval())

# calculate loss using model.evaluate
loss2 = model.evaluate(x=x, y=y_true)
print("loss2 = ", loss2)

# show loss using model.fit
model.fit(x=x, y=y_true)
```
The result is:
```
loss1 =  14.506287
convolution engine.
10/10 [==============================] - 0s 5ms/step
loss2 =  14.50628662109375
Epoch 1/1
10/10 [==============================] - 0s 21ms/step - loss: 14.5063
```

Any repiies will be appreciated.
Thanks.

**Comment by sanatmpa1** [comment]
@River861,

I am able to reproduce the issue in colab with TF 2.6 and here is the [gist](https://colab.research.google.com/gist/sanatmpa1/58d9505fc6f1db5942a44a34b1eb7c09/51720.ipynb)

I can see that this issue is mostly related to Keras component and so can you please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues),To know more see;[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)

**Comment by River861** [comment]
@sanatmpa1 ,
Thanks for your response.
